pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -132,10 +137,11 @@ const (
 	retrySleepTime = 20 * time.Millisecond
 )
 
-type nodeStatusData struct {
+type nodeHealthData struct {
 	probeTimestamp           metav1.Time
 	readyTransitionTimestamp metav1.Time
 	status                   v1.NodeStatus
```

Comment:
Since you're touching this code anyway - can we change this to pointer? [potentially in a separate PR if that's easier - you can do all the renames there too]

Reply:
After reading this PR more, if you could move just renames to a separate PR, that would make this PR so much easier.
Thanks!